packages/flutter/test/rendering/sliver_fixed_extent_layout_test.dart

```diff
@@ -41,6 +41,42 @@ void main() {
     expect(children[1].attached, false);
     expect(children[2].attached, true);
   });
+
+  group('getMaxChildIndexForScrollOffset', () {
+    const double itemExtent = 600.0;
+    testValidationRounding(1234, 0, 0, 'When item extent is 0');
+    testValidationRounding(0, itemExtent, 0, 'First index when offset 0');
+    testValidationRounding(itemExtent, itemExtent, 0,
+        'First index when offset is equal to item extent');
+    testValidationRounding(itemExtent + 1, itemExtent, 1,
+        'Second index when offset is greater than item extent');
+    testValidationRounding(itemExtent + 0.0000001, itemExtent, 1,
+        'Second index when offset is slightly greater than item extent');
+    testValidationRounding(
+        itemExtent * 4.5, itemExtent, 4, 'Moving to the 4 index');
+
+    testValidationRounding(
+        2468.571428571429, 411.42857142857144, 5, 'Edge case for Android');
+    testValidationRounding(
+        2484.0, 414.0, 5, 'Same Android edge case but for iOS');
```

Comment:
I do not understand how these tests are testing different Android and iOS behavior. How is the platform factored into this?

Reply:
Actually, it's not because of the platform, it's because of the screen size, the itemExtent property. I'll change the description